web/quickfeed_service.go

```
@@ -367,7 +399,18 @@ func (s *QuickFeedService) GetSubmissionsByCourse(_ context.Context, in *connect
 
 // UpdateSubmission is called to approve the given submission or to undo approval.
 func (s *QuickFeedService) UpdateSubmission(_ context.Context, in *connect.Request[qf.UpdateSubmissionRequest]) (*connect.Response[qf.Void], error) {
-	err := s.updateSubmission(in.Msg.GetSubmissionID(), in.Msg.GetGrades(), in.Msg.GetReleased(), in.Msg.GetScore())
+	submission, err := s.db.GetSubmission(&qf.Submission{ID: in.Msg.GetSubmissionID()})
+	if err != nil {
+		return nil, connect.NewError(connect.CodeNotFound, errors.New("failed to get submission"))
```

We should log the actual error on the server here, like we do below (Line 415):
```go
s.logger.Errorf("UpdateSubmission failed: %v", err)
```
We don't want to return the actual error to the user, but we want to have a log entry if it does happen... Can help debugging a problem in the future.

---

Ping @meling

---

Do you have a question about this? Or are you looking for guidance on what to write in the error messages? If so, write what you think is appropriate and I will suggest improvements in code review.

---

My main point is that we should always log the root cause error, even if we also send it to the frontend. We always want to log errors. They happen rarely, so they don’t cause any major concerns in the logs. But when they happen they may be essential for debugging a problem.

---

I think you misunderstood my confusion by;

> "We don't want to return the actual error to the user"

Can you clarify if this contradicts with the current errors returned in the api end-points or if there's something im missing here?